File: VB/MySpellCheckerModule/SpellCheckerHelper.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraEditors
Imports DevExpress.ExpressApp.Win.Editors
Imports DevExpress.XtraSpellChecker

Namespace MySpellCheckerModule
	Public Class LargeStringEditTextBoxFinder
		Inherits DevExpress.XtraSpellChecker.Native.MemoEditTextBoxFinder
		Public Overrides Function GetTextBoxInstance(ByVal editControl As Control) As TextBoxBase
			If TypeOf editControl Is LargeStringEdit Then
				Return MyBase.GetTextBoxInstance(CType(editControl, MemoEdit))
			End If
			Return Nothing
		End Function
	End Class
	Public Class StringEditTextBoxFinder
		Inherits DevExpress.XtraSpellChecker.Native.TextEditTextBoxFinder
		Public Overrides Function GetTextBoxInstance(ByVal editControl As Control) As TextBoxBase
			If TypeOf editControl Is StringEdit Then
				Return MyBase.GetTextBoxInstance(CType(editControl, TextEdit))
			End If
			Return Nothing
		End Function
	End Class
	Public NotInheritable Class SpellCheckerHelper
		Private Shared dictionary_Renamed As SpellCheckerISpellDictionary = Nothing
		Private Shared customDictionary_Renamed As SpellCheckerCustomDictionary = Nothing
		Private Sub New()
		End Sub
		Shared Sub New()
			dictionary_Renamed = New SpellCheckerISpellDictionary()
			dictionary_Renamed.AlphabetPath = "Dictionaries\EnglishAlphabet.txt"
			dictionary_Renamed.DictionaryPath = "Dictionaries\American.xlg"
			dictionary_Renamed.GrammarPath = "Dictionaries\English.aff"
			dictionary_Renamed.CaseSensitive = False
			dictionary_Renamed.Culture = New System.Globalization.CultureInfo("en-US")
			dictionary_Renamed.Encoding = Encoding.GetEncoding(1252)
			dictionary_Renamed.Load()
			customDictionary_Renamed = New SpellCheckerCustomDictionary()
			customDictionary_Renamed.AlphabetPath = dictionary_Renamed.AlphabetPath
			customDictionary_Renamed.DictionaryPath = "Dictionaries\Custom.txt"
			customDictionary_Renamed.CaseSensitive = False
			customDictionary_Renamed.Culture = dictionary_Renamed.Culture
			customDictionary_Renamed.Encoding = dictionary_Renamed.Encoding
			customDictionary_Renamed.Load()
			DevExpress.XtraSpellChecker.Native.SpellCheckTextControllersManager.Default.RegisterClass(GetType(StringEdit), GetType(DevExpress.XtraSpellChecker.Native.SimpleTextEditTextController))
			DevExpress.XtraSpellChecker.Native.SpellCheckTextControllersManager.Default.RegisterClass(GetType(LargeStringEdit), GetType(DevExpress.XtraSpellChecker.Native.SimpleTextEditTextController))
			DevExpress.XtraSpellChecker.Native.SpellCheckTextControllersManager.Default.RegisterClass(GetType(MemoEdit), GetType(DevExpress.XtraSpellChecker.Native.SimpleTextEditTextController))
			DevExpress.XtraSpellChecker.Native.SpellCheckTextBoxBaseFinderManager.Default.RegisterClass(GetType(StringEdit), GetType(StringEditTextBoxFinder))
			DevExpress.XtraSpellChecker.Native.SpellCheckTextBoxBaseFinderManager.Default.RegisterClass(GetType(LargeStringEdit), GetType(LargeStringEditTextBoxFinder))
		End Sub
		Public Shared ReadOnly Property Dictionary() As SpellCheckerISpellDictionary
			Get
				Return dictionary_Renamed
			End Get
		End Property
		Public Shared ReadOnly Property CustomDictionary() As SpellCheckerCustomDictionary
			Get
				Return customDictionary_Renamed
			End Get
		End Property
	End Class
End Namespace

File: CS/MySpellCheckerModule/SpellCheckerHelper.cs
using System;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraEditors;
using DevExpress.XtraSpellChecker;
using DevExpress.ExpressApp.Win.Editors;

namespace MySpellCheckerModule {
    public class LargeStringEditTextBoxFinder : DevExpress.XtraSpellChecker.Native.MemoEditTextBoxFinder {
        public override TextBoxBase GetTextBoxInstance(Control editControl) {
            if (editControl is LargeStringEdit)
                return base.GetTextBoxInstance((MemoEdit)editControl);
            return null;
        }
    }
    public class StringEditTextBoxFinder : DevExpress.XtraSpellChecker.Native.TextEditTextBoxFinder {
        public override TextBoxBase GetTextBoxInstance(Control editControl) {
            if (editControl is StringEdit)
                return base.GetTextBoxInstance((TextEdit)editControl);
            return null;
        }
    }
    public static class SpellCheckerHelper {
        private static SpellCheckerISpellDictionary dictionary;
        private static SpellCheckerCustomDictionary customDictionary;
        static SpellCheckerHelper() {
            dictionary = new SpellCheckerISpellDictionary();
            dictionary.AlphabetPath = "Dictionaries\\EnglishAlphabet.txt";
            dictionary.DictionaryPath = "Dictionaries\\American.xlg";
            dictionary.GrammarPath = "Dictionaries\\English.aff";
            dictionary.CaseSensitive = false;
            dictionary.Culture = new System.Globalization.CultureInfo("en-US");
            dictionary.Encoding = Encoding.GetEncoding(1252);
            dictionary.Load();
            customDictionary = new SpellCheckerCustomDictionary();
            customDictionary.AlphabetPath = dictionary.AlphabetPath;
            customDictionary.DictionaryPath = "Dictionaries\\Custom.txt";
            customDictionary.CaseSensitive = false;
            customDictionary.Culture = dictionary.Culture;
            customDictionary.Encoding = dictionary.Encoding;
            customDictionary.Load();
            DevExpress.XtraSpellChecker.Native.SpellCheckTextControllersManager.Default.RegisterClass(
                typeof(StringEdit), typeof(DevExpress.XtraSpellChecker.Native.SimpleTextEditTextController));
            DevExpress.XtraSpellChecker.Native.SpellCheckTextControllersManager.Default.RegisterClass(
                typeof(LargeStringEdit), typeof(DevExpress.XtraSpellChecker.Native.SimpleTextEditTextController));
            DevExpress.XtraSpellChecker.Native.SpellCheckTextControllersManager.Default.RegisterClass(
                typeof(MemoEdit), typeof(DevExpress.XtraSpellChecker.Native.SimpleTextEditTextController));
            DevExpress.XtraSpellChecker.Native.SpellCheckTextBoxBaseFinderManager.Default.RegisterClass(
                typeof(StringEdit), typeof(StringEditTextBoxFinder));
            DevExpress.XtraSpellChecker.Native.SpellCheckTextBoxBaseFinderManager.Default.RegisterClass(
                typeof(LargeStringEdit), typeof(LargeStringEditTextBoxFinder));
        }
        public static SpellCheckerISpellDictionary Dictionary {
            get { return dictionary; }
        }
        public static SpellCheckerCustomDictionary CustomDictionary {
            get { return customDictionary; }
        }
    }
}
